### PR TITLE
sRGB conversion fix

### PR DESCRIPTION
**Pull request by Crisspl**
Regarding PR #273

### CODE REVIEW

**Comment by manhnt9** [comment]
Thank you. Could you also take a look at this issue? https://github.com/buildaworldnet/IrrlichtBAW/issues/294 I'm trying to debug an app.

**Comment by Crisspl** [comment]
Seems like missing -lunwind, but never got this before..
Btw you can open GitHub PR (or just pull with Git) from any fork to yours any time you want yourself (so that you don't have to wait for someone to make PR)

**Comment by manhnt9** [comment]
Have you tried `cmake .. -DCMAKE_BUILD_TYPE=Debug` on Linux? If this error is not here before it means the CMake is really not stable, I've met similar issue with zlib in irrbaw. Before it linked and after a while it doesn't.

**Comment by Crisspl** [comment]
Hm, no, I think I ran only Release builds on Linux.

File: include/irr/video/decodePixels.h [hunk 1]
```diff
@@ -947,172 +947,72 @@ namespace irr { namespace video
...
     inline void decodePixels<asset::EF_R8_SRGB, double>(const void* _pix[4], double* _output, uint32_t _blockX, uint32_t _blockY)
     {
         const uint8_t& pix = reinterpret_cast<const uint8_t*>(_pix[0])[0];
-        _output[0] = ((pix >> 0) & 0xffULL) / 255.;
-        {
-            double& lin = _output[0];
-            if (lin <= 0.4045) lin /= 12.92;
-            else lin = pow((lin + 0.055) / 1.055, 2.4);
-        };
+        _output[0] = impl::srgb2lin(((pix >> 0) & 0xffULL) / 255.);
```
Comment (devshgraphicsprogramming) [inline include/irr/video/decodePixels.h:963]: @Crisspl what changed?  how does this fix the bug?

**Comment by devshgraphicsprogramming** [comment]
I changed base to sRGB branch since thats what we're working on now

@manhnt9 @Florastamine this PR fixes:
- linking of libunwind Debug builds of examples on Linux
- srgb conversions in decode and encode pixels improved and fixed (At least partially)
- minor bugfixes

**Comment by devshgraphicsprogramming** [comment]
@manhnt9 note that this moves where the pch on Linux go (build dir, not source dir) so you can remove the .gch files from gitignore that you've added

**Comment by Crisspl** [comment]
>note that this moves where the pch on Linux go (build dir, not source dir) so you can remove the .gch files from gitignore that you've added

Actually it was done in https://github.com/buildaworldnet/IrrlichtBAW/commit/3227a97354cd6893136022c526e285798e273843 , long before this PR

**Comment by manhnt9** [comment]
Updated.

**Pull request merged**